pkg/monitoring/metrics/common/client/latency_adapter.go

```diff
@@ -31,5 +32,29 @@ type latencyAdapter struct {
 }
 
 func (l *latencyAdapter) Observe(_ context.Context, verb string, u url.URL, latency time.Duration) {
-	l.m.WithLabelValues(verb, u.String()).Observe(latency.Seconds())
+	l.m.WithLabelValues(determineVerb(u, verb), u.String()).Observe(latency.Seconds())
+}
+
+func determineVerb(u url.URL, verb string) string {
+	if verb != "GET" {
+		return verb
+	}
+
+	if strings.Contains(u.Path, "/watch/") || u.Query().Get("watch") == "true" {
+		return "WATCH"
+	}
+
+	resource := findResource(u)
+	if resource == "" {
+		return "none"
```

Comment:
if the resource is "", should this `return verb`?

Reply:
No, in the case of url like `https://10.223.0.1:443/api?limit=10`, it is returning none